### PR TITLE
feat(client): management

### PR DESCRIPTION
**Pull request by mafewtm**
## Description

Adds job management capabilities

## Checklist

- [ ] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: client/main.lua [hunk 1]
```diff
@@ -41,11 +41,43 @@ local function createDuty(job, station)
...
+            coords = location.coords,
+            radius = location.radius or 1.5,
+            debug = config.debugPoly,
+            options = {
+                {
+                    name = ('%sBossMenu'):format(job),
+                    icon = 'fa-solid fa-people-roof',
+                    label = 'Open Job Management',
+                    canInteract = function()
+                        return QBX.PlayerData.job.isboss and QBX.PlayerData.job.onduty
```
Comment (ArturMichalak) [inline client/main.lua:62]: Are you sure that only the Chief of Police can use this menu?

File: client/main.lua [hunk 2]
```diff
@@ -41,11 +41,43 @@ local function createDuty(job, station)
...
+
 CreateThread(function()
     Wait(150)
 
     for job, data in pairs(sharedConfig.departments) do
         createBlip(data.blip)
+        createManagement(job, data.management)
         createDuty(job, data.duty)
     end
 end)
```
Comment (ArturMichalak) [inline client/main.lua:83]: Please keep the last line blank, it prevents false changes

**Comment by ArturMichalak** [comment]
How about adding an ox_target variant?

**Review by ArturMichalak — APPROVED** [review]

File: config/shared.lua [hunk 3]
```diff
@@ -26,6 +32,13 @@ return {
                 scale = 0.8,
                 color = 29
             },
+            management = {
+                {
+                    coords = vec3(447.04, -974.01, 30.44),
+                    radius = 1.5,
+                    groups = { police = 0 }
```
Comment (ArturMichalak) [inline config/shared.lua:39]: An array can be used. https://overextended.dev/ox_target/Options

File: client/main.lua (hunk 2 above)
Comment (mafewtm) [reply client/main.lua:83]: Our editorconfig removes the last line upon save. No new line is needed. No one should be submitting PRs with nothing changed anyways

File: config/shared.lua (hunk 3 above)
Comment (mafewtm) [reply config/shared.lua:39]: huh?

Comment (ArturMichalak) [reply config/shared.lua:39]: If the grand level is unused, you can write it like this
groups = {'police'}

Comment (ArturMichalak) [reply config/shared.lua:39]: https://github.com/overextended/ox_target/blob/aefc464d01da9b7aa3565e79161dd0a489945b90/client/framework/qb.lua#L55-L75

Comment (mafewtm) [reply config/shared.lua:39]: I am doing this on purpose so that people can edit the grade as they see fit.

Comment (ArturMichalak) [reply config/shared.lua:39]: In that case, "isboss" should also be configurable

Comment (mafewtm) [reply config/shared.lua:39]: it is. `qbx_core/shared/jobs.lua`

Comment (ArturMichalak) [reply config/shared.lua:39]: I mean, configurable with qbx_policejob.

Comment (mafewtm) [reply config/shared.lua:39]: Why would it be configurable within policejob?

Comment (ArturMichalak) [reply config/shared.lua:39]: to improve user experience

Comment (mafewtm) [reply config/shared.lua:39]: That's not how that works

File: client/main.lua (hunk 2 above)
Comment (ArturMichalak) [reply client/main.lua:83]: When the last line break is missing and something is added to the end of the file, the last line of the old file will be falsely highlighted as changed. This is the main reason why I encourage to insert it when I see such an opportunity.
https://github.com/hedge-code/no-eol-at-eof/pull/4/files

File: config/shared.lua [hunk 4]
```diff
@@ -7,13 +7,19 @@ return {
...
 
+    ---@class Management
+    ---@field coords vector3
+    ---@field groups table
+    ---@field radius? number radius of the zone. default is 1.5
+
     ---@class Duty
     ---@field coords vector3
     ---@field groups table
     ---@field radius? number radius of the zone. default is 1.5
```
Comment (Manason) [inline config/shared.lua:18]: These classes have the same fields. Should we merge them to be one type?

File: config/shared.lua (hunk 3 above)
Comment (Manason) [reply config/shared.lua:39]: qbx_management only allows bosses to open the management menu. https://github.com/Qbox-project/qbx_management/blob/c1743020832fa28a8af9d931c8e243eb3e468018/client/main.lua#L138

So we should just get rid of the groups field here and instead require that the player is a police boss to show them this.

**Review by Manason — CHANGES_REQUESTED** [review]

File: config/shared.lua (hunk 4 above)
Comment (mafewtm) [reply config/shared.lua:18]: Later

File: config/shared.lua (hunk 3 above)
Comment (mafewtm) [reply config/shared.lua:39]: It's like that so that no one whos a boss of another job can access the point

Comment (Manason) [reply config/shared.lua:39]: Why is it a config option though? What are you wanting server owners to put? It seems like we already have the info we need without asking server owners.

**Pull request merged**